app/code/core/Mage/ConfigurableSwatches/Block/Catalog/Media/Js/List.php

```diff
@@ -64,4 +64,19 @@ protected function _getImageSizes()
     {
         return ['small_image'];
     }
+
+    /**
+     * Prevent actual block render if we are disabled, and i.e. via the module
+     * config as opposed to the advanced module settings page
+     *
+     * @return string
+     */
+    protected function _toHtml()
+    {
+        // No listview attribute configured
+        if (!Mage::getStoreConfigFlag('configswatches/general/product_list_attribute')) {
```

Comment:
```suggestion
        if (!Mage::getStoreConfigFlag('configswatches/general/product_list_attribute')) {
```
since the "if" condition it's self explanatory